src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -266,16 +261,34 @@ public unsafe BitArray And(BitArray value)
 
             int count = GetInt32ArrayLengthFromBitLength(Length);
 
+            // Unroll loop for count less than Vector256 size.
             switch (count)
             {
+                case 7: m_array[6] &= value.m_array[6]; goto case 6;
+                case 6: m_array[5] &= value.m_array[5]; goto case 5;
+                case 5: m_array[4] &= value.m_array[4]; goto case 4;
+                case 4: m_array[3] &= value.m_array[3]; goto case 3;
                 case 3: m_array[2] &= value.m_array[2]; goto case 2;
                 case 2: m_array[1] &= value.m_array[1]; goto case 1;
                 case 1: m_array[0] &= value.m_array[0]; goto Done;
                 case 0: goto Done;
             }
 
             int i = 0;
-            if (Sse2.IsSupported)
+            if (Avx2.IsSupported)
```

Comment:
Do we know if `BitArray` is commonly large enough that we won't regress perf for the 16-32 byte scenario since `Avx2` will force that back to the scalar case?

It's likely worth determining the cutoff point at which AVX2 starts significantly winning out over SSE2 and account for that (I have seen it generally be at 48-128 bytes, rather than at exactly 32-bytes).